### DISCUSSION TITLE
Proper vertex snapping from the 21st century

### DISCUSSION THREAD

**Discussion by Cosmotect** [godotengine/godot-proposals#10359] [Editor]
Hello! I am here to advocate for a feature that will end up saving many millenia of human time in the editor. Good vertex snapping is an incredibly important tool for efficient level design.

Unreal despite its massive strengths completely drops the ball in this regard, vertex snapping is unusable there.

However, Unity, has undeniably the BEST vertex snapping solution out of any 3D editor to ever grace the planet. Just look at the video: Around the 10 second mark I hold down "V" which forces the selected object's pivot to float around with the cursor simultaneously snapping to the closest vertex. Then, around the 17 second mark, I press "LMB" (still holding "V") which picks up the object from the last place the pivot had moved to, and while dragging the object around in this fashion, it now snaps to all other vertecies in the scene with the vertex you picked it up with.

I am years away from being able to contribute this to the engine. I am hoping that the community sees the sheer BRILLIANCE of this feature, and adds it to the engine.

Thank you for your attention

https://github.com/user-attachments/assets/dda6af36-d684-477a-9353-151a373f2810

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/755.

**Reply by Cosmotect** [reply]
Even though vertex snapping is a big part of it, the floating pivot toggle is also huge, please keep that in mind when consolidating with other vertex snapping requests.
Thanks!

**Reply by Calinou** [reply]
Thanks for the reminder &ndash; I've posted a comment in the proposal accordingly.
